agenta-backend/agenta_backend/migrations/mongo_to_postgres/utils.py

```diff
@@ -0,0 +1,155 @@
+        mapping = IDsMappingDB(
+            id=id, table_name=table_name, objectid=str(mongo_id), uuid=uuid
+        )
+        session.add(mapping)
+        await session.commit()
+
+
+async def get_mapped_uuid(mongo_id):
+    """Retrieve the mapped UUID for a given MongoDB ObjectId."""
+    async with db_engine.get_session() as session:
+        stmt = select(IDsMappingDB.uuid).filter(IDsMappingDB.objectid == str(mongo_id))
+        result = await session.execute(stmt)
+        row = result.first()
+        return row[0] if row else None
+
+
+def get_datetime(value):
+    """Helper function to handle datetime fields."""
+    if isinstance(value, str):
+        return datetime.fromisoformat(value.replace("Z", "+00:00"))
+    return value if value else datetime.now(timezone.utc)
+
+
+def generate_uuid():
+    """Generate a new UUID."""
+    return uuid.uuid7()
+
+
+def update_migration_report(collection_name, total_docs, migrated_docs):
+    migration_report[collection_name] = {"total": total_docs, "migrated": migrated_docs}
+
+
+def print_migration_report():
+    print("\n====================== Migration Report ======================")
+
+    # Headers
+    headers = ["Table", "Total in MongoDB", "Migrated to PostgreSQL"]
+
+    if not migration_report:
+        print("No data available in the migration report.")
+        return
+
+    # Determine the maximum lengths for each column including headers
+    max_table_length = max(
+        len(headers[0]), max(len(table) for table in migration_report.keys())
+    )
+    max_total_length = max(
+        len(headers[1]),
+        max(len(str(counts["total"])) for counts in migration_report.values()),
+    )
+    max_migrated_length = max(
+        len(headers[2]),
+        max(len(str(counts["migrated"])) for counts in migration_report.values()),
+    )
+
+    # Set the header and divider with appropriate padding
+    table_header = f"| {headers[0].ljust(max_table_length)} | {headers[1].ljust(max_total_length)} | {headers[2].ljust(max_migrated_length)} |"
+    table_divider = f"|{'-' * (max_table_length + 2)}|{'-' * (max_total_length + 2)}|{'-' * (max_migrated_length + 2)}|"
+
+    print(table_header)
+    print(table_divider)
+
+    for table, counts in migration_report.items():
+        table_row = f"| {table.ljust(max_table_length)} | {str(counts['total']).ljust(max_total_length)} | {str(counts['migrated']).ljust(max_migrated_length)} |"
+        print(table_row)
+
+
+async def migrate_collection(
+    collection_name, model_class, transformation_func, association_model=None
+):
+    """General function to migrate a collection to a SQL table."""
+    print(
+        f"\n====================== Migrating {collection_name}... ======================\n"
+    )
+    total_docs = mongo_db[collection_name].count_documents({})
+    migrated_docs = 0
+
+    async with db_engine.get_session() as session:
+        for skip in range(0, total_docs, BATCH_SIZE):
```

Comment:
I'd use here tqdm (especially for cloud) to get some overview how things are going

```suggestion
        for skip in tqdm(range(0, total_docs, BATCH_SIZE), total=(total_docs-1)//BATCH_SIZE + 1, desc=f"Migrating {collection_name}"):
```

Reply:
nice lib thanks